core/src/test/scala/org/apache/spark/executor/ExecutorSuite.scala

```diff
@@ -402,6 +403,77 @@ class ExecutorSuite extends SparkFunSuite
     assert(taskMetrics.getMetricValue("JVMHeapMemory") > 0)
   }
 
+  test("SPARK-32175: Plugin initialization should start after heartbeater started") {
+    val tempDir = Utils.createTempDir()
+
+    val importStatements =
+      """
+        |import java.util.Map;
+        |import org.apache.spark.api.plugin.*;
```

Comment:
What about using the qualified class name to avoid adding the new parameter `preClassDefinitionBlock`?

Reply:
Yeah, it's better.